dotCMS/src/main/java/com/dotmarketing/db/commands/UpsertCommand.java

```diff
@@ -0,0 +1,306 @@
+package com.dotmarketing.db.commands;
+
+import com.dotcms.repackage.org.apache.commons.lang.StringUtils;
+import com.dotcms.system.SimpleMapAppContext;
+import com.dotmarketing.common.db.DotConnect;
+import com.dotmarketing.common.util.SQLUtil;
+import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.db.DbType;
+import com.dotmarketing.exception.DotDataException;
+import com.liferay.util.StringUtil;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Generic UpsertCommand that can be used to Generate and execute Native SQL Upsert Queries
+ * @author Andre Curione
+ */
+public abstract class UpsertCommand implements DatabaseCommand {
+
+    /**
+     * Map that contains Upsert Command implementations
+     */
+    private static Map<DbType, UpsertCommand> commandMap = new ConcurrentHashMap<>();
+
+    static {
+        commandMap.put(DbType.H2, new H2UpsertCommand());
+        commandMap.put(DbType.POSTGRESQL, new PostgreUpsertCommand());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this reference to "PostgreUpsertCommand". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2390)